lib/gitlib_test.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+#
+# gitlib.sh unit tests
+#
+source $(dirname $(readlink -ne $BASH_SOURCE))/common.sh
+source $TOOL_LIB_PATH/gitlib.sh
+source $TOOL_LIB_PATH/releaselib.sh
+
+##############################################################################
+# TESTING VER_REGEX regex's
+##############################################################################
+echo "Testing VER_REGEX:"
+echo
+# Ex. v1.5.0-alpha.2.435+8c67d08e3a535d
+DOTZERO="v1.4.0"
+SHORT="v1.4.5"
+LONG="v1.4.5-alpha.0"
+BUILD="v1.4.5-alpha.0.435"
```

Comment:
is this actually a version we'll ever see? I think if you are getting the build number, `git describe` (which is what hack/lib/version.sh uses) will also always append the sha.